### PR TITLE
Add explanatory reference to ``PiecewiseChebychev``

### PR DESCRIPTION
**Pull request by Cryoris**
and update other locations with published version

### Summary

Add a missing reference to `PiecewiseChebychev` that explains where the circuit is coming from. Also update locations of with the arXiv version with the published version of the paper.

### CODE REVIEW

File: qiskit/algorithms/linear_solvers/hhl.py [hunk 1]
```diff
@@ -87,12 +87,12 @@ class HHL(LinearSolver):
...
         [1]: Harrow, A. W., Hassidim, A., Lloyd, S. (2009).
-        Quantum algorithm for linear systems of equations.
-        `Phys. Rev. Lett. 103, 15 (2009), 1–15. <https://doi.org/10.1103/PhysRevLett.103.150502>`_
+             Quantum algorithm for linear systems of equations.
+             `Phys. Rev. Lett. 103, 15 (2009), 1–15. <https://doi.org/10.1103/PhysRevLett.103.150502>`_
 
-        [2]: Carrera Vazquez, A., Hiptmair, R., & Woerner, S. (2020).
-        Enhancing the Quantum Linear Systems Algorithm using Richardson Extrapolation.
-        `arXiv:2009.04484 <http://arxiv.org/abs/2009.04484>`_
+        [2]: Vazquez, A. C., Hiptmair, H., & Woerner, S. (2022).
```
Comment (anedumla) [inline qiskit/algorithms/linear_solvers/hhl.py:93]: Carrera Vazquez, A.

File: qiskit/circuit/library/arithmetic/piecewise_chebyshev.py [hunk 2]
```diff
@@ -51,6 +52,9 @@ class PiecewiseChebyshev(BlueprintCircuit):
         [1]: Haener, T., Roetteler, M., & Svore, K. M. (2018).
              Optimizing Quantum Circuits for Arithmetic.
              `arXiv:1805.12445 <http://arxiv.org/abs/1805.12445>`_
+        [2]: Vazquez, A. C., Hiptmair, H., & Woerner, S. (2022).
```
Comment (anedumla) [inline qiskit/circuit/library/arithmetic/piecewise_chebyshev.py:55]: Carrera Vazquez, A.

**Review by anedumla — COMMENTED** [review]
looks good, just that Carrera is my surname :)

File: qiskit/algorithms/linear_solvers/hhl.py [hunk 3]
```diff
@@ -87,12 +87,12 @@ class HHL(LinearSolver):
...
-        `Phys. Rev. Lett. 103, 15 (2009), 1–15. <https://doi.org/10.1103/PhysRevLett.103.150502>`_
+             Quantum algorithm for linear systems of equations.
+             `Phys. Rev. Lett. 103, 15 (2009), 1–15. <https://doi.org/10.1103/PhysRevLett.103.150502>`_
 
-        [2]: Carrera Vazquez, A., Hiptmair, R., & Woerner, S. (2020).
-        Enhancing the Quantum Linear Systems Algorithm using Richardson Extrapolation.
-        `arXiv:2009.04484 <http://arxiv.org/abs/2009.04484>`_
+        [2]: Carrera Vazquez, A., Hiptmair, R., & Woerner, S. (2022).
+             Enhancing the Quantum Linear Systems Algorithm Using Richardson Extrapolation.
+             `ACM Transactions on Quantum Computing 3, 1, Article 2 https://doi.org/10.1145/3490631`_
```
Comment (mtreinish) [inline qiskit/algorithms/linear_solvers/hhl.py:95]:
```suggestion
             `ACM Transactions on Quantum Computing 3, 1, Article 2 <https://doi.org/10.1145/3490631>`_
```

File: qiskit/circuit/library/arithmetic/piecewise_polynomial_pauli_rotations.py [hunk 4]
```diff
@@ -81,9 +81,9 @@ class PiecewisePolynomialPauliRotations(FunctionalPauliRotations):
              Optimizing Quantum Circuits for Arithmetic.
              `arXiv:1805.12445 <http://arxiv.org/abs/1805.12445>`_
 
-        [2]: Carrera Vazquez, A., Hiptmair, R., & Woerner, S. (2020).
+        [2]: Carrera Vazquez, A., Hiptmair, R., & Woerner, S. (2022).
              Enhancing the Quantum Linear Systems Algorithm using Richardson Extrapolation.
-             `arXiv:2009.04484 <http://arxiv.org/abs/2009.04484>`_
+             `ACM Transactions on Quantum Computing 3, 1, Article 2 https://doi.org/10.1145/3490631`_
```
Comment (mtreinish) [inline qiskit/circuit/library/arithmetic/piecewise_polynomial_pauli_rotations.py:86]:
```suggestion
             `ACM Transactions on Quantum Computing 3, 1, Article 2 <https://doi.org/10.1145/3490631>`_
```

File: qiskit/circuit/library/arithmetic/piecewise_chebyshev.py [hunk 5]
```diff
@@ -51,6 +52,9 @@ class PiecewiseChebyshev(BlueprintCircuit):
         [1]: Haener, T., Roetteler, M., & Svore, K. M. (2018).
              Optimizing Quantum Circuits for Arithmetic.
              `arXiv:1805.12445 <http://arxiv.org/abs/1805.12445>`_
+        [2]: Carrera Vazquez, A., Hiptmair, H., & Woerner, S. (2022).
+             Enhancing the Quantum Linear Systems Algorithm Using Richardson Extrapolation.
+             `ACM Transactions on Quantum Computing 3, 1, Article 2 https://doi.org/10.1145/3490631`_
```
Comment (mtreinish) [inline qiskit/circuit/library/arithmetic/piecewise_chebyshev.py:57]:
```suggestion
             `ACM Transactions on Quantum Computing 3, 1, Article 2 <https://doi.org/10.1145/3490631>`_
```

**Review by mtreinish — COMMENTED** [review]
The docs builds are failing because of a syntax error with the external links

**Review by mtreinish — APPROVED** [review]

**Pull request merged**